WooCommerce/Classes/ViewRelated/Dashboard/Blaze/BlazeCampaignDashboardView.swift

```diff
@@ -40,6 +54,44 @@ final class BlazeCampaignDashboardViewHostingController: SelfSizingHostingContro
 }
 
 private extension BlazeCampaignDashboardViewHostingController {
+    private func createBlazeNavigationController() {
+        blazeNavigationController = UINavigationController(rootViewController: createProductSelectorViewController())
+    }
+
+    private func createProductSelectorViewController() -> ProductSelectorViewController {
+        let productSelectorViewModel = ProductSelectorViewModel(
+            siteID: viewModel.siteID,
+            purchasableItemsOnly: true,
+            onProductSelectionStateChanged: { [weak self] product in
+                guard let self = self else { return }
+
+                // Navigate to Campaign Creation Form once a product is selected.
+                let campaignCreationFormViewModel = BlazeCampaignCreationFormViewModel(siteID: viewModel.siteID,
```

Comment:
BlazeCampaignCreationForm itself does not support creating from an existing product yet. Later on we should pass product id here.